### ISSUE TITLE
Can't resolve str split function signature correctly

### ISSUE THREAD

**Issue by spindensity** [microsoft/pyright#165]
Pyright reports an incompatible type error in the following code. Anyway, the code is correct.

```python
import os


def split_para_type(path_str):
    return os.path.basename(path_str).split(".")
```

![1](https://user-images.githubusercontent.com/33576822/61073801-c585ed80-a448-11e9-929d-a2228b0766c2.png)
![2](https://user-images.githubusercontent.com/33576822/61073822-d20a4600-a448-11e9-98ed-b49e66bd9eb8.png)

Add type annotation to `split_para_type` parameter can avoid this error:
![3](https://user-images.githubusercontent.com/33576822/61073933-10076a00-a449-11e9-8e6b-7c8c8d678d96.png)

**Comment by erictraut** [comment]
The behavior is correct from a type checking standpoint. The function `os.path.basename` returns either a str or a bytes result. Pyright can disambiguate between the two return types based on the input type. However, in your case, you haven't provided an input type (`path_str` is untyped), so pyright assumes that the output is either str or bytes (`Union[str, bytes]`). The split method must take an input parameter that matches the type. If `path_str` happens to be `bytes`, your code will generate an exception, and this is what pyright is flagging.

To fix this, you can simply add a type annotation to path_str to tell pyright that it's always a str and never bytes (`path_str: str`).
